src/commitment/pedersen/constraints.rs

```diff
@@ -62,28 +62,26 @@ where
     ) -> Result<Self::OutputVar, SynthesisError> {
         assert!((input.len() * 8) <= (W::WINDOW_SIZE * W::NUM_WINDOWS));
 
-        let mut padded_input = input.to_vec();
-        // Pad if input length is less than `W::WINDOW_SIZE * W::NUM_WINDOWS`.
-        if (input.len() * 8) < W::WINDOW_SIZE * W::NUM_WINDOWS {
-            let current_length = input.len();
-            for _ in current_length..((W::WINDOW_SIZE * W::NUM_WINDOWS) / 8) {
-                padded_input.push(UInt8::constant(0u8));
-            }
-        }
-
-        assert_eq!(padded_input.len() * 8, W::WINDOW_SIZE * W::NUM_WINDOWS);
-        assert_eq!(parameters.params.generators.len(), W::NUM_WINDOWS);
-
-        // Allocate new variable for commitment output.
-        let input_in_bits: Vec<Boolean<_>> = padded_input
+        // Convert input bytes to little-endian bits
+        let mut input_in_bits: Vec<Boolean<_>> = input
             .iter()
             .flat_map(|byte| byte.to_bits_le().unwrap())
             .collect();
+
+        // Pad input to `W::WINDOW_SIZE * W::NUM_WINDOWS`.
+        let padding_size = (W::WINDOW_SIZE * W::NUM_WINDOWS) - input_in_bits.len();
```

Comment:
maybe we should make this a `checked_sub`, and error if `input_in_bits.len() > W::WINDOW_SIZE * W::NUM_WINDOWS`?